ledger-service/db-backend/src/main/scala/com/digitalasset/http/dbbackend/Queries.scala

```diff
@@ -406,6 +410,7 @@ object Queries {
       contractId: String,
       templateId: TpId,
       key: CK,
+      keyHash: Option[String],
```

Comment:
I don't think this needs to be incorporated in `DBContract`.

`Queries` already contains a mechanism for dealing with this in a backend-dependent way: `type DBContractKey`. You can change the related conversion function to incorporate the hash calculation you've added to `ContractsFetch`. I know you have it that early because the function doesn't work on `JsValue`, per se, but I think this can be straightforwardly dealt with by changing the constraints on the `CK` type parameter on `Queries#insertContracts` (i.e. type `CK` must have both an associated `JsonWriter` _and_ a means to calculate the hash, the latter of which may be unused for Postgres).